dedupe/convenience.py

```diff
@@ -24,55 +24,94 @@ def consoleLabel(deduper): # pragma: no cover
     '''
 
     finished = False
+    has_previous = False
+    use_previous = False
     fields = unique(field.field
                     for field
                     in deduper.data_model.primary_fields)
 
 
-    while not finished :
-        n_match, n_distinct = (len(deduper.training_pairs['match']),
-                               len(deduper.training_pairs['distinct']))
-
-        uncertain_pairs = deduper.uncertainPairs()
-
+    buffer_len = 1 # Max number of previous operations
+    examples_buffer = []
+    record_pair_buffer = [] 
+
+    while not finished :     
+            
+        if has_previous and use_previous:
+            record_pair = record_pair_buffer[0]
+            record_pair_buffer = record_pair_buffer[1:]
+            examples_buffer = examples_buffer[1:]
+            use_previous = False
+        else:
+            uncertain_pairs = deduper.uncertainPairs()
+            assert len(uncertain_pairs) == 1 
+            record_pair = uncertain_pairs[0]
+       
+        n_match_in_buffer = sum(len(x['match']) for x in examples_buffer)
+        n_distinct_in_buffer = sum(len(x['distinct']) for x in examples_buffer)
+                                                                                        
+        n_match, n_distinct = (len(deduper.training_pairs['match']) + n_match_in_buffer,
+                        len(deduper.training_pairs['distinct']) + n_distinct_in_buffer)
+      
         examples = {'distinct' : [], 'match' : []}
 
-        for record_pair in uncertain_pairs:
-            label = ''
-            labeled = False
-
-            for pair in record_pair:
-                for field in fields:
-                    line = "%s : %s" % (field, pair[field])
-                    print(line, file=sys.stderr)
-                print(file=sys.stderr) 
-
-            print("{0}/10 positive, {1}/10 negative".format(n_match, n_distinct),
-                  file=sys.stderr)
-            print('Do these records refer to the same thing?', file=sys.stderr)
-            valid_response = False
-            while not valid_response:
+        label = ''
+        labeled = False
+
+        for pair in record_pair:
+            for field in fields:
+                line = "%s : %s" % (field, pair[field])
+                print(line, file=sys.stderr)
+            print(file=sys.stderr) 
+
+        print("{0}/10 positive, {1}/10 negative".format(n_match, n_distinct),
+                file=sys.stderr)
+        print('Do these records refer to the same thing?', file=sys.stderr)
+        valid_response = False
+        while not valid_response:
+            if has_previous:
```

Comment:
make the prompt and valid responses variables and you can reduce the code duplication here.